sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -218,7 +247,14 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
       bucketSpec = getBucketSpec,
       options = extraOptions.toMap)
 
-    dataSource.write(mode, df)
+    val destination = source match {
+      case "jdbc" => extraOptions.get(JDBCOptions.JDBC_TABLE_NAME)
+      case _ => extraOptions.get("path")
```

Comment:
What other parameters are you thinking about?

It's pretty easy to wrap this class in some other class that has these output params and any other params you want to expose, but it would be nice to understand exactly what you're referring to here.